assignments/pull_requests.go

```diff
@@ -0,0 +1,157 @@
+package assignments
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	pb "github.com/autograde/quickfeed/ag"
+	"github.com/autograde/quickfeed/database"
+	"github.com/autograde/quickfeed/kit/score"
+	"github.com/autograde/quickfeed/scm"
+)
+
+var (
+	// These are used to track how many times someone has been assigned to review a pull request. They map as follows.
+	// teacherReviewCounter[courseID][userID] = count
+	// groupReviewCounter[groupID][userID] = count
+	teacherReviewCounter = make(map[uint64]map[uint64]int)
```

Comment:
Likely not critical, but this counter will reset if we have to restart the server for some reason.

Also, would it be an idea to decrement this counter for a user when a PR they are reviewing is done? This way we could assign PRs to TAs who have the least amount of outstanding PRs.
This would require us to store the ID of the reviewer in pull requests.

This could prove useful in scenarios where one TA is assigned PRs that are resolved quickly, while another TA is assigned some which requires more work relative to the other.
(However, these scenarios could arise not due to a high workload for the TA, but rather slow response time from students, or some other reason.)

We could also, if useful, use this to display an overview of PRs, and reviewers, in the frontend.

Implementing this will require database lookups.

Reply:
I guess it depends on how complex you want this feature to be, but I am definetly on board if so.